### DISCUSSION TITLE
Possible bug when placing a subject sink in a different branch

### DISCUSSION THREAD

**Discussion by ZegCricket** [bonsai-rx/bonsai#1799] [General]
Hello! I hope you are doing well.

I'm opening this discussion to report a possible bug (although this bug appears when coding the workflow in the wrong way). It's difficult for me to describe what happens by words alone, so I'll show a piece of a Bonsai workflow to illustrate the possible bug.
![bug1](https://github.com/bonsai-rx/bonsai/assets/105512708/fb450f96-3fc8-402c-9e81-a678cf6ea6eb)
If, from the Bonsai workflow above, I disconnect the MulticastSubject node from the Zip node and connect the Add node to the Zip instead, an additional and identical MulticastSubject node appears at the bottom, as shown:
![bug2](https://github.com/bonsai-rx/bonsai/assets/105512708/e7af46ed-9cb2-4e12-b382-2e6b8e379855)
If I delete one of the MulticastSubject nodes, the two nodes get deleted. However, if I connected the MulticastSubject after the Add node was connected to the Zip, this bug will not happen.

I know this is a minor bug which is easily solved by properly coding the workflow (as shown in the first figure), but I thought it could be of your interest anyway.

Thank you for your attention!

**Comment by glopesdev** [comment]
@ZegCricket can you attach a copy of the initial workflow, before you perform the action you describe causing the node duplication? That would be very useful.

**Reply by ZegCricket** [reply]
Of course! The workflow I'm sending is already a modified version of a workflow I'm working on. The figures from the original post correspond to the StartTrial SelectMany node.
[bug_catcher.zip](https://github.com/bonsai-rx/bonsai/files/15374763/bug_catcher.zip)
